src/deprecated.jl

```diff
@@ -0,0 +1,10 @@
+using Base: depwarn
+
+const INF_WARN = Ref(false)
+
+function timetype_inf_warning()
+    if !INF_WARN[]
+        depwarn("`infinity` support for Dates.TimeType is deprecated. Use `InfExtendedTime{T}` instead", :timetype_inf_warning)
```

Comment:
Interpolating should allow the types to be fully qualified depending on what the user has imported. Possibly you may want to do `$(InfExtendedTime){T}` depending on what you want displayed.
```suggestion
        depwarn("`infinity` support for $(Dates.TimeType) is deprecated. Use `$InfExtendedTime` instead", :timetype_inf_warning)
```